Build error on espidf v5

Hello,

when trying to build the master branch with espidf, I get this error:
```
lib/ESP32-HUB75-MatrixPanel-DMA/src/platforms/esp32/esp32_i2s_parallel_dma.hpp:65:1: error: 'i2s_dev_t' does not name a type; did you mean 'i2s_dir_t'?
   65 | i2s_dev_t* getDev();
      | ^~~~~~~~~
      | i2s_dir_t
lib/ESP32-HUB75-MatrixPanel-DMA/src/platforms/esp32/esp32_i2s_parallel_dma.hpp:148:14: error: 'i2s_dev_t' does not name a type; did you mean 'i2s_dir_t'?
  148 |     volatile i2s_dev_t* _dev;
      |              ^~~~~~~~~
      |              i2s_dir_t
```

Adding
```cpp
#include <soc/i2s_struct.h>
```
to `esp32_i2s_parallel_dma.hpp` fixes it, but makes new errors come up:
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/30153207/054feafb-37f7-4ae9-ab7e-345af4ee73d1)

Those errors can be fixed by adding
```cpp
#include <soc/i2s_periph.h>
```
to `esp32_i2s_parallel_dma.cpp` and changing the `%d` format specifiers to `%ld`.

--- a/src/platforms/esp32/esp32_i2s_parallel_dma.cpp
+++ b/src/platforms/esp32/esp32_i2s_parallel_dma.cpp
@@ -27,6 +27,7 @@ Modified heavily for the ESP32 HUB75 DMA library by:
 #include <driver/gpio.h>
 #include <driver/periph_ctrl.h>
 #include <soc/gpio_sig_map.h>
+#include <soc/i2s_periph.h>
 
 #if defined (ARDUINO_ARCH_ESP32)
 #include <Arduino.h>
@@ -199,7 +200,7 @@ Modified heavily for the ESP32 HUB75 DMA library by:
     ////////////////////////////// Clock configuration //////////////////////////////
 
     auto freq 		= (_cfg.bus_freq);
-    ESP_LOGD("ESP32/S2", "Requested output clock frequency: %d Mhz",  (freq/1000000));   		
+    ESP_LOGD("ESP32/S2", "Requested output clock frequency: %ld Mhz",  (freq/1000000));   		
 	
 	// What is the current CPU frequency?
 
@@ -487,7 +488,7 @@ Modified heavily for the ESP32 HUB75 DMA library by:
     if ( !dmadesc_b )
     {
       if ( (_dmadesc_a_idx+1) > _dmadesc_count) {
-        ESP_LOGE("ESP32/S2", "Attempted to create more DMA descriptors than allocated memory for. Expecting a maximum of %d DMA descriptors", _dmadesc_count);          
+        ESP_LOGE("ESP32/S2", "Attempted to create more DMA descriptors than allocated memory for. Expecting a maximum of %ld DMA descriptors", _dmadesc_count);          
         return;
       }
     }

--- a/src/platforms/esp32/esp32_i2s_parallel_dma.hpp
+++ b/src/platforms/esp32/esp32_i2s_parallel_dma.hpp
@@ -44,6 +44,7 @@ Contributors:
 #include <sys/types.h>
 #include <freertos/FreeRTOS.h>
 #include <driver/i2s.h>
+#include <soc/i2s_struct.h>
 #include <rom/lldesc.h>
 #include <rom/gpio.h>
 

